File: cs50x/cash.py
while True:
    cents = input("Change owed: ")
    try:
        cents_float = float(cents)
        if cents_float > 0:
            break
        else:
            print("Invalid! Enter a positive amount.")
    except ValueError:
        print("Invalid! Enter a numeric value.")

# convert dollars to cents (integer)
cents_int = round(cents_float * 100)

# calculate coins
quarters = cents_int // 25
cents_remaining = cents_int % 25

dimes = cents_remaining // 10
cents_remaining = cents_remaining % 10

nickels = cents_remaining // 5
cents_remaining = cents_remaining % 5

pennies = cents_remaining  # remaining cents are pennies

total_coins = quarters + dimes + nickels + pennies
print(total_coins)

File: cs50x/cash.c
#include <cs50.h>
#include <stdio.h>

int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // prompt the user to change owed
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    // Calculate the number of quarters and update cents
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes and update cents
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels and update cents
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies and update cents
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum the number of coins
    int total_coins = quarters + dimes + nickels + pennies;

    // Print the total number of coins
    printf("%i\n", total_coins);
}
int calculate_quarters(int cents)
{
    // Calculate how many quarters you should give customer
    int quarters = 0;
    while (cents >= 25)
    {
        quarters++;
        cents = cents - 25;
    }
    return quarters;
}
// Calculate how many dimes you should give customer

int calculate_dimes(int cents)
{
    int dimes = 0;
    while (cents >= 10)
    {
        dimes++;
        cents = cents - 10;
    }
    return dimes;
}
// Calculate how many nickels you should give customer
int calculate_nickels(int cents)
{
    int nickels = 0;
    while (cents >= 5)
    {
        nickels++;
        cents = cents - 5;
    }
    return nickels;
}
// calculate how many pennies you should give customer
int calculate_pennies(int cents)
{
    int pennies = 0;
    while (cents >= 1)
    {
        pennies++;
        cents = cents - 1;
    }
    return pennies;
}
